src/3rdparty/walkontable/src/selection.js

```diff
@@ -180,109 +197,105 @@ class Selection {
     };
   }
 
+  addClassAtElem(elem, classNames) {
+    if (elem) {
+      addClass(elem, classNames);
+    }
+  }
+
   /**
    * @param wotInstance
    */
-  draw(wotInstance) {
+  draw(wotInstance, selectedCellFn) {
     if (this.isEmpty()) {
-      if (this.settings.border) {
-        this.getBorder(wotInstance).disappear();
+      if (this.hasSelectionHandle()) {
+        const found = this.getSelectionHandleIfExists(wotInstance);
+        if (found) {
+          found.disappear();
+        }
       }
 
       return;
     }
 
     const renderedRows = wotInstance.wtTable.getRenderedRowsCount();
-    const renderedColumns = wotInstance.wtTable.getRenderedColumnsCount();
-    const corners = this.getCorners();
-    const [topRow, topColumn, bottomRow, bottomColumn] = corners;
-
-    for (let column = 0; column < renderedColumns; column += 1) {
-      const sourceCol = wotInstance.wtTable.columnFilter.renderedToSource(column);
+    if (renderedRows === 0) {
+      return;
+    }
 
-      if (sourceCol >= topColumn && sourceCol <= bottomColumn) {
-        const TH = wotInstance.wtTable.getColumnHeader(sourceCol);
+    const renderedColumns = wotInstance.wtTable.getRenderedColumnsCount();
+    if (renderedColumns === 0) {
+      return;
+    }
 
-        if (TH) {
-          const newClasses = [];
+    const corners = this.getCorners();
+    const [firstRow, firstColumn, lastRow, lastColumn] = corners;
 
-          if (this.settings.highlightHeaderClassName) {
-            newClasses.push(this.settings.highlightHeaderClassName);
-          }
+    if (this.settings.highlightHeaderClassName || this.settings.highlightColumnClassName) {
+      for (let sourceColumn = firstColumn; sourceColumn <= lastColumn; sourceColumn += 1) {
+        this.addClassAtElem(wotInstance.wtTable.getColumnHeader(sourceColumn), this.settings.highlightHeaderClassName, this.settings.highlightColumnClassName);
```

Comment:
Maybe you could cache classNames and add them once at the end?

Reply:
This method was refactored and I think this comment is not applicable anymore. Current source: https://github.com/handsontable/handsontable/blob/51cfaa911af785051e3840578a904e5159bf54c5/src/3rdparty/walkontable/src/selection.js#L234-L284

Any suggestions?